django/contrib/admin/options.py

```diff
@@ -1408,9 +1411,14 @@ def changeform_view(self, request, object_id=None, form_url='', extra_context=No
             errors=helpers.AdminErrorList(form, formsets),
             preserved_filters=self.get_preserved_filters(request),
         )
-
+        # Hide "Save" and "Save and continue" buttons since "Save as New" was
+        # previously chosen and the interface gets confusing otherwise.
+        # Also, the "normal" save functionality is broken after a "Save as New"
```

Comment:
I think we could probably omit this second sentence. At least, try to make the comments detailed enough to avoid referencing tickets in code.